packages/core/src/browser/progress-bar.ts

```diff
@@ -27,7 +27,7 @@ export class ProgressBar implements Disposable {
 
     protected progressBar: HTMLDivElement;
 
-    constructor(protected options: ProgressBar.Options, onProgress: Event<ProgressLocationEvent>) {
+    constructor(protected options: ProgressBar.Options, onProgress: Event<LocationProgress>) {
```

Comment:
We should read initial the progress state in the constructor before listening to incoming.

It would be better to be `ProgressBar` injectable via `ProgressBarFactory` accepting options and providing a new progress bar then we could inject `ProgressBar.Options` and `ProgressService` inside `ProgressBar`. Also It would be aligned wit our coding guidelines to use DI instead of `new` operator for extensibility.